lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -872,7 +872,17 @@ def on_tls_conf_set(
 
         # In case of renewal of the unit transport layer cert - restart opensearch
         if renewal and self.is_admin_user_configured():
+            if self.peers_data.get(Scope.UNIT, "tls_ca_renewed", False):
```

Comment:
Only way I could find to pass this [line](https://github.com/canonical/opensearch-operator/blob/1de95171ff6688d3a290176f33d84d20ab6cdda3/lib/charms/opensearch/v0/opensearch_tls.py#L656) check